lib/LibraryTemplatePlugin.js

```diff
@@ -23,7 +23,14 @@ const accessorToObjectAccess = accessor => {
  * @returns {string} the path
  */
 const accessorAccess = (base, accessor, joinWith = "; ") => {
-	const accessors = Array.isArray(accessor) ? accessor : [accessor];
+	let accessors;
+	if (Array.isArray(accessor)) {
+		accessors = accessor;
+	} else if (typeof accessor === "function") {
+		accessors = [accessor()];
+	} else {
+		accessors = [accessor];
+	}
```

Comment:
hey @sokra, I attempted to allow functions for lib names first. In the case of `SetVarMainTemplatePlugin` it's not possible to evaluate the function inside the plugin because in most of the cases this function goes as a first parameter, modifying `name` property provided by `LibraryTemplatePlugin`.

Do you feel like we should move this functionality inside `SetVarMainTemplatePlugin` since it's not used in other plugins?

Reply:
You don't have to change this function:

Instead of

``` js
 					new SetVarMainTemplatePlugin(
 						`var ${accessorAccess(undefined, this.name)}`,
 						false
 					).apply(compilation);
```


``` js
 					new SetVarMainTemplatePlugin(
 						typeof this.name === "function" ?
 							data => `var ${accessorAccess(undefined, this.name(data))}` :
 							`var ${accessorAccess(undefined, this.name)}`,
 						false
 					).apply(compilation);
```

Reply:
Ok, I see your point. But isn't it gonna produce a lot of duplicate code? Perhaps I'm missing something but I don't understand why it has to be done this way.

Reply:
Instead of duplicating code you could move this code into a method:

``` js
_withName(fn) {
  return typeof this.name === "function" ?
    data => fn(this.name(data)) :
    fn(this.name)
}

new SetVarMainTemplatePlugin(
  this._withName(name => `var ${accessorAccess(undefined, name)}`)
  false
).apply(compilation);